modules/dnn/src/opencl/gemm_buffer.cl

```diff
@@ -510,7 +510,7 @@ __kernel void TEMPLATE(gemm_buffer_NT, Dtype)(
         Dtype8 arow;
 
 #define READ_BROW(_brow, _row) \
-        _brow = as_half8(vload4(0, (__local float *)(slm_brow0 + _row * SLM_BLOCK))); \
+        _brow = vload8(0, slm_brow0 + _row * SLM_BLOCK); \
```

Comment:
Current test is `24x5` by `6x5`. Need to add a test with bigger dimensions to go to an inner loop.